### PR TITLE
UIOR-263 Create primary search for POLines

### PR DESCRIPTION
**Pull request by AndreiShumski2**
## Purpose
Create the primary search options for Order lines
https://issues.folio.org/browse/UIOR-263
## Approach
Add searchableIndexes to `OrderLinesList`
## Screenshot
uior263
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/OrderLinesList/OrderLinesList.js [hunk 1]
```diff
@@ -20,7 +20,10 @@ import {
   VENDORS,
 } from '../components/Utils/resources';
 // import OrderLinesFilters from './OrderLinesFilters';
-// import { filterConfig } from './OrdersLinesFilterConfig';
+import {
+  // filterConfig,
```
Comment (aliaksei-chumakou) [inline src/OrderLinesList/OrderLinesList.js]: just keep commented line as is, to reduce number of conflicts

Comment (AndreiShumski2) [reply src/OrderLinesList/OrderLinesList.js]: Got it. Fixed

**Comment by AndreiShumski2** [comment]
> should we care about search all?
In [UIOR-273](https://issues.folio.org/browse/UIOR-273) we have to add primary search option of Title + Contributor Order lines

File: src/OrderLinesList/OrderLinesList.js [hunk 2]
```diff
@@ -59,13 +62,24 @@ const columnWidths = {
   funCodes: '18%',
 };
 
+const queryTemplate = `(
```
Comment (aliaksei-chumakou) [inline src/OrderLinesList/OrderLinesList.js]: `queryTemplate` is a very general name, could you please make a more specific name?

File: src/OrderLinesList/OrdersLinesFilterConfig.js [hunk 3]
```diff
@@ -50,3 +49,13 @@ export const filterConfig = [
     values: [],
   },
 ];
+
+export const searchableIndexes = [
+  { label: 'Keyword', value: '', makeQuery: term => `(title="${term}*" or contributors="${term}*" or poLineNumber="${term}*" or requester="${term}*" or vendorDetail.vendorAccount="${term}*" or vendorDetail.refNumber="${term}*" or details.productIds="${term}*")` },
```
Comment (aliaksei-chumakou) [inline src/OrderLinesList/OrdersLinesFilterConfig.js]: Are these labels translated? I suppose all strings should be tranlsated

File: src/OrderLinesList/OrderLinesList.js [hunk 4]
```diff
@@ -25,8 +25,11 @@ import {
   FUNDS,
 } from '../components/Utils/resources';
 import OrderLinesFilters from './OrderLinesFilters';
-import { filterConfig } from './OrdersLinesFilterConfig';
 import Details from './Details';
+import {
+  filterConfig,
+  searchableIndexes,
```
Comment (NikitaSedyx) [inline src/OrderLinesList/OrderLinesList.js]: you you please move `searchableIndexes` to another file?  e.g. `OrdersLinesSearchConfig`

File: src/OrderLinesList/OrdersLinesFilterConfig.js (hunk 3 above)
Comment (NikitaSedyx) [reply src/OrderLinesList/OrdersLinesFilterConfig.js]: and please format object

Comment (AndreiShumski2) [reply src/OrderLinesList/OrdersLinesFilterConfig.js]: fixed

File: src/OrderLinesList/OrderLinesList.js (hunk 2 above)
Comment (AndreiShumski2) [reply src/OrderLinesList/OrderLinesList.js]: Updated

File: src/OrderLinesList/OrdersLinesSearchConfig.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
+// eslint-disable-next-line import/prefer-default-export
+export const searchableIndexes = [
+  {
+    label: 'keyword',
```
Comment (NikitaSedyx) [inline src/OrderLinesList/OrdersLinesSearchConfig.js]: can it be `FormattedMessage` instead of translation in component?

Comment (AndreiShumski2) [reply src/OrderLinesList/OrdersLinesSearchConfig.js]: Unfortunately, no. In this case, you will get [Object Object] in options

**Pull request merged**